AP_HAL_ChibiOS: Update readme for RADIX2HD

mluessi opened the pull request:
Updated the readme for the RADIX2HD target with better instruction for how to obtain or create firmware files.

Also changed SERIAL4 to ESC telemetry, as this is now the default.

mluessi commented:
@Hwurzburg this should be helpful for the wiki page. Please let me know if anything is unclear.

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/RADIX2HD/README.md:
```diff
@@ -86,29 +86,66 @@ In addition to voltage sensing, the board also has an input for an external curr
+### Option 2: Download the Firmware From the ArduPilot Build Server
+
+Download the ELF file from the [ArduPilot Build Server](https://firmware.ardupilot.org/).
+Make sure you download the file for the "RADIX2HD" target. The ELF file for ArduCopter
```
```suggestion
Make sure you download the file for the "RADIX2HD" target. For example, the ELF file for ArduCopter
```

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/RADIX2HD/README.md:
```diff
@@ -86,29 +86,66 @@ In addition to voltage sensing, the board also has an input for an external curr
+
+Download the ELF file from the [ArduPilot Build Server](https://firmware.ardupilot.org/).
+Make sure you download the file for the "RADIX2HD" target. The ELF file for ArduCopter
+is called "arducopter.elf".
```
```suggestion
is called "arducopter.elf". At this time only the 4.5 ("latest") versions are available.
```

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/RADIX2HD/README.md:
```diff
@@ -86,29 +86,66 @@ In addition to voltage sensing, the board also has an input for an external curr
+
+    pip install https://github.com/BrainFPV/brainfpv_fw_packer/archive/main.zip
+
+After installing it, you can use the following command to convert the "arducopter.elf"
```
```suggestion
After installing it, you can use the following command to convert the "arducopter.elf" (or other vehicle elf file)
```

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/RADIX2HD/README.md:
```diff
@@ -86,29 +86,66 @@ In addition to voltage sensing, the board also has an input for an external curr
 
     ./waf configure --board RADIX2HD
     ./waf copter
 
```
```suggestion
other vehicles can be built, but the RADIX2HD is targetted primarily for copter applications.
```

Hwurzburg reviewed (changes requested):
yes...looking good..a few nits//..but the download site still has no ArduPilot firmware so until that is in place, I cant post the wiki...but will work on the wip in the meantime

mluessi commented:
@Hwurzburg  thanks for the comments. I included them.

Hwurzburg commented:
@mluessi you need to squash to one commit for the AP_HAL_ChibiOS library

also, what uart RX is the DJI connector's "SBUS" pin really connected to? figure show UART7, but there is no UART7

mluessi commented:
@Hwurzburg please excuse the delay. I just went over this again, made some edits and squashed the commits. 

Note that support for it is now included in the 4.4 version, so I removed the part where it said it was only supported in 4.5. 

Support for UART7 is being added in #24485

peterbarker commented:
@Hwurzburg OK to merge this?
